tests/components/sunweg/test_sensor.py

```diff
@@ -0,0 +1,185 @@
+"""Tests for the Sun WEG sensor."""
+
+from freezegun.api import FrozenDateTimeFactory
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.sunweg.const import DOMAIN
+from homeassistant.components.sunweg.coordinator import SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .common import SUNWEG_MOCK_ENTRY
+
+from tests.common import async_fire_time_changed
+
+
+async def test_sensor_total(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    freezer: FrozenDateTimeFactory,
+    api_fixture,
+    plant_fixture_alternative,
+) -> None:
+    """Test sensor type total."""
+    mock_entry = SUNWEG_MOCK_ENTRY
+    mock_entry.add_to_hass(hass)
+
+    assert await async_setup_component(hass, DOMAIN, mock_entry.data)
+    freezer.tick(SCAN_INTERVAL)
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    state = hass.states.get("sensor.plant_123_total_money_lifetime")
+    assert state.state == snapshot
+    assert state.attributes == snapshot
+
+    state = hass.states.get("sensor.plant_123_total_energy_today")
+    assert state.state == snapshot
+    assert state.attributes == snapshot
+
+    state = hass.states.get("sensor.plant_123_total_output_power")
+    assert state.state == snapshot
+    assert state.attributes == snapshot
+
+    state = hass.states.get("sensor.plant_123_total_lifetime_energy_output")
+    assert state.state == snapshot
+    assert state.attributes == snapshot
+
+    state = hass.states.get("sensor.plant_123_total_last_update")
+    assert state.state == snapshot
+    assert state.attributes == snapshot
+
+    api_fixture["plant"].return_value = plant_fixture_alternative
+
+    freezer.tick(SCAN_INTERVAL)
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    state = hass.states.get("sensor.plant_123_total_last_update")
+    assert state.state == snapshot
+    assert state.attributes == snapshot
```

Comment:
I would recommend to do this in a separate test, and don't use snapshots here as it hides what you are testing